airbyte-cdk/java/airbyte-cdk/db-sources/src/main/java/io/airbyte/cdk/integrations/debezium/internals/postgres/PostgresReplicationConnection.java

```diff
@@ -48,7 +48,7 @@ public static Connection createConnection(final JsonNode jdbcConfig) throws SQLE
       validateReplicationConnection(connection);
       return connection;
     } catch (final PSQLException exception) {
-      if (exception.getMessage().equals("FATAL: must be superuser or replication role to start walsender")) {
+      if ("42501".equals(exception.getSQLState())) { // insufficient_privilege
```

Comment:
The existing logic doesn't work for postgres 16. In any case we shouldn't have business logic depending on textual messages, as far as postgres is concerned the sql state values are the proper API to use.